src/main/topLevelPages/myDashboard/newVirtualContributorWizard/useNewVirtualContributorWizard.tsx

```diff
@@ -0,0 +1,216 @@
+import { Dialog } from '@mui/material';
+import { ComponentType, useCallback, useMemo, useState } from 'react';
+import CreateNewVirtualContributorStep0 from './CreateNewVirtualContributorStep0';
+import CreateSubspaceStep1 from './CreateSubspaceStep1';
+import ChooseSubspaceStep1b from './ChooseSubspaceStep1b';
+import {
+  useCreateSubspaceMutation,
+  useCreateVirtualContributorOnAccountMutation,
+  useNewVirtualContributorMySpacesQuery,
+} from '../../../../core/apollo/generated/apollo-hooks';
+import {
+  AiPersonaBodyOfKnowledgeType,
+  AiPersonaDataAccessMode,
+  AiPersonaEngine,
+  SpaceType,
+} from '../../../../core/apollo/generated/graphql-schema';
+import NameVirtualContributorStep2 from './NameVirtualContributorStep2';
+import WhatsNextStep3 from './WhatsNextStep3';
+import { useTranslation } from 'react-i18next';
+import { useNotification } from '../../../../core/ui/notifications/useNotification';
+
+enum Step {
+  step0,
+  step1,
+  step1b,
+  step2,
+  step3,
+}
+
+interface SelectedSubspace {
+  id: string;
+  profile: {
+    displayName: string;
+    url: string;
+  };
+}
+
+interface useNewVirtualContributorWizardProvided {
+  startWizard: () => void;
+  NewVirtualContributorWizard: ComponentType<NewVirtualContributorWizardProps>;
+}
+
+interface NewVirtualContributorWizardProps {}
+
+const useNewVirtualContributorWizard = (): useNewVirtualContributorWizardProvided => {
+  const { t } = useTranslation();
+  const notify = useNotification();
+
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [step, setStep] = useState<Step>(Step.step0);
+  const onDialogClose = () => {
+    setDialogOpen(false);
+    setStep(Step.step0);
+  };
+
+  const { data, loading } = useNewVirtualContributorMySpacesQuery();
+
+  const { mySpaceId, myAccountId, mySpaceName, mySubspaces, bokSubspaces } = useMemo(() => {
+    const mySpace = data?.me.mySpaces[0]; // TODO: For now we just take the first space
```

Comment:
Here we should filter by account.host.id === currentUser.id. In mySpaces there are space you're part of but you're not the host of.

Reply:
the filter in the graphql would be sufficient.